### DISCUSSION TITLE
BirdDog NDI Encoder/Decoder Recipe

### DISCUSSION THREAD

**Discussion by dargs** [museumsvictoria/nodel#319] [Recipes]
Before I get too deep doing my own, has anyone else already made a recipe to control BirdDog NDI encoder/decoders ? 

https://birddog.tv/studio-overview/
https://birddog.tv/mini-overview/

ACMI has a handful of units but have only really used them a few times and I'm still working out best way of having them on our network, without causing any issues (maybe on their own vlan etc)

When they work, they've been ok at sending/receiving video from our studio to green room spaces, or as a way to send the output of a Scala player to multiple locations. We also have vMix, ZoomISO, Zoom Rooms, Panasonic PTZs etc that can rx/tx NDI.

The Companion module works quite well..  https://github.com/bitfocus/companion-module-birddog-converters/ 

I got as far as doing some get_url commands, getting data ok, but posting it to make changes is tbc if it's working... 

```

Get Encoder Settings [http://xxxx:8080/enc-settings]
Response [{"ndiaudio":"unmute","ndisendgn":"BirdDog","ndisendgnen":"ndisendgnend","nditally":"tallyon","ndivideoq":"120","ndivideoqsel":"ndivideoqc"}]

Get About [http://xxxx:8080/about]
Response: [{"manufacturer":"BirdDog","DeviceType":"BirdDog","Version":"1.0","MyHostName":"birddog-e9809"}]

Get AV Settings [http://xxxx:8080/av-settings]
Response[{"ainputsel":"ason","ajingain":"50","ajoutgain":"75","ajoutput":"ajoutputd","audinsel":"audinsele","avtallyh":"avtallyoffh","avtallys":"avtallyoffs","hvideoincs":"hvideoincsrgb","videoin":"AUTO","videoincs":"rgb","videoout":"videooutd","videoouth":"avoshdmim","videoouts":"avossdim","vidinsel":"hdmi"}]

Get Source List [http://xxxx:8080/List]
Response: [{"BIRDDOG-1C34E (comms)":"xxxx:5961","BIRDDOG-1C34E (SDI)":"xxxx:5962","BIRDDOG-E9809 (comms)":"xxxx:5961"}]
```

**Comment by justparking** [comment]
Hey @dargs,

I forgot to mention that the Museum Victoria do actually make use of the Minis.

It's a primative recipe for the encoders (status monitoring) and can turn a collection of decoders into a matrix switcher.

See https://github.com/automaticau/nodel-recipes/blob/newdevice_birddog-mini/BirdDog%20Mini%20Encoder%20or%20Decoder/script.py

It was written a very long time ago.

**Reply by dargs** [reply]
Oh cool! thanks @justparking  !  Works for my one decoder on the network at the moment! 

I'm curious how the remote source list event might work.  In my test code, I could Get a Source list from the decoder itself, and was considering putting that into a list to choose from, but remotely managing the source list for multiple decoders makes more sense. 

![image](https://github.com/museumsvictoria/nodel/assets/1365481/257ad686-052a-4f29-a324-905e25aa0de1)

Now I know the MV is using them, I'm also wondering about how peoples networking is going using these fairly bandwidth intensive devices? are they on their own vlan etc ?
